emop-maven-plugin/src/main/java/edu/cornell/RppMojo.java

```diff
@@ -48,29 +57,69 @@ private void invokeSurefire() throws MojoExecutionException {
         }
     }
 
+    public static void writeSpecsToFile(Set<String> specs, File file) throws FileNotFoundException {
+        try (PrintWriter writer = new PrintWriter(file)) {
+            for (String spec : specs) {
+                writer.println(spec);
+            }
+        }
+    }
+
+    public void updateCriticalAndBackgroundSpecs(String criticalViolationsPath, String bgViolationsPath, String javamopAgent)
+            throws MojoExecutionException, FileNotFoundException {
+        Set<String> criticalSpecsSet = new HashSet<>();
+        Set<String> allSpecs = Util.retrieveSpecListFromJar(javamopAgent, getLog());
+        if (!demoteCritical && !RppHandlerMojo.criticalSpecsSet.equals(allSpecs)) {
+            // need to demote specs if the critical specs set contained all specs (first run)
```

Comment:
I do not see how this demotion is being done.

Reply:
The second clause (`!RppHandlerMojo.criticalSpecsSet.equals(allSpecs)`) of line 72 specifies that if the set of specs run in the critical phase is the set of all specs, we will not execute the then block, i.e. keep any specs that were not violated in our set of critical specs. 

The code after the block basically only adds specs that were violated to the set of critical specs. This then block ensures that specs that were previously in the set of critical specs will stay.

Reply:
Sorry if my question was not clear:

`// need to demote specs`

Where is that demotion happening?

Reply:
Demotion happens implicitly; the set of critical specs gets updated on line [86](https://github.com/SoftEngResearch/emop/pull/43/files#diff-fe2c6c2966ccc47e497a8d650a143ccc170a66b4382d5f6bc56769827bfab650R86) with the set of violated specs (across both critical and background runs). Any spec that doesn't make that cut is "demoted".

Reply:
Oh, I see.. this PR is for the case where we demote specs that were not violated in SHA `N` to the background when we monitor SHA `N+1`? If so, please add a comment about the implicit demotion to line 86.

Will we see a future PR for demoting specs every `X` SHAs, instead of every SHA?